subsys/pm/pm_ctrl.c

```diff
@@ -18,7 +18,7 @@ LOG_MODULE_DECLARE(power);
 
 static atomic_t power_state_disable_count[PM_STATES_LEN];
 
-void pm_constraint_set(enum pm_state state)
+__weak void pm_constraint_set(enum pm_state state)
```

Comment:
nit: this should go into a separate commit, since from now on constraint can be overridden.

Since usage of `__weak` is now being debated, an alternative could be a Kconfig option like `CONFIG_PM_CUSTOM_CONSTRAINTS`, selected by SoC code. See https://github.com/zephyrproject-rtos/zephyr/pull/37435#issuecomment-916564329

Reply:
I was aware of it. The problem is that we have weak linkage all over the project and not a clear a plan to get rid of this. Probably instead of adding more KConfig symbols that can easily became a mess, we should add some kind of PM API that can be implemented by platform and applications and have some well defined precedence.
This solution will not get into 2.7 release, so instead of adding an extra KConfig symbol that will need to lead with later this fix is simpler and has less impact.

Of course, if  the direction to go regarding weak symbol is that, I definitely can change it.  My only point it that this seems to over complicate it.

Reply:
If adding the weak symbol fixes the issue for 2.7.0, then i think it's fine to add. Does it need to be in a separate commit?